README.adoc

```diff
@@ -16,6 +16,45 @@ SmallRye Fault Tolerance is an implementation of {microprofile-fault-tolerance}[
 SmallRye Fault Tolerance provides some additional APIs on top of MicroProfile Fault Tolerance.
 Before proper documentation exists, please refer to the `api` directory directly.
 
+==== @Blocking and @NonBlocking
+
+In addition to the MicroProfile Fault Tolerance `@Asynchronous` annotation, which can be placed on methods returning `Future` or `CompletionStage`, SmallRye Fault Tolerance also supports 2 more annotations for asynchronous processing:
+
+* `@io.smallrye.common.annotation.Blocking`
+* `@io.smallrye.common.annotation.NonBlocking`
+
+These annotations can have multiple meanings, depending on context.
+SmallRye Fault Tolerance only pays attention to these annotations if:
+
+* they are placed on methods that return `CompletionStage` (the `Future` type can't really be used for non-blocking processing);
+* they are placed on methods that apply some fault tolerance strategy (such as `@Fallback`, defined either on a method or a class).
+
+Under these circumstances, SmallRye Fault Tolerance assigns these annotations the following meaning:
+
+* `@Blocking` means that execution of the operation will be offloaded to another thread (that's what `@Asynchronous` does by default);
```

Comment:
I would add something like: `Uses this annotation if you don't want to block the caller thread and the annotated method has blocking logic.`